File: core/src/main/resources/org/treblereel/gwt/three4g/closure/material/MeshDistanceMaterialParameters.js
import { MaterialParameters } from './MaterialParameters.js';

import { Texture } from '../../textures/Texture.js';

/**
 * parameters = {
 *
 *  referencePosition: <float>,
 *  nearDistance: <float>,
 *  farDistance: <float>,
 *
 *  skinning: <bool>,
 *  morphTargets: <bool>,
 *
 *  map: new THREE.Texture( <Image> ),
 *
 *  alphaMap: new THREE.Texture( <Image> ),
 *
 *  displacementMap: new THREE.Texture( <Image> ),
 *  displacementScale: <float>,
 *  displacementBias: <float>
 *
 * }
 */

/**
 * @record
 * @extends {MaterialParameters}
 */
var MeshDistanceMaterialParameters = function() {};
/** @type {number|undefined} */
MeshDistanceMaterialParameters.prototype.referencePosition;
/** @type {number|undefined} */
MeshDistanceMaterialParameters.prototype.nearDistance;
/** @type {number|undefined} */
MeshDistanceMaterialParameters.prototype.farDistance;

/** @type {boolean|undefined} */
MeshDistanceMaterialParameters.prototype.skinning;
/** @type {boolean|undefined} */
MeshDistanceMaterialParameters.prototype.morphTargets;

/** @type {Texture|undefined} */
MeshDistanceMaterialParameters.prototype.map;

/** @type {Texture|undefined} */
MeshDistanceMaterialParameters.prototype.alphaMap;

/** @type {Texture|undefined} */
MeshDistanceMaterialParameters.prototype.displacementMap;
/** @type {number|undefined} */
MeshDistanceMaterialParameters.prototype.displacementScale;
/** @type {number|undefined} */
MeshDistanceMaterialParameters.prototype.displacementBias;



export { MeshDistanceMaterialParameters };

File: core/src/main/java/org/treblereel/gwt/three4g/materials/MeshDistanceMaterialParameters.java
package org.treblereel.gwt.three4g.materials;

import jsinterop.annotations.JsOverlay;
import jsinterop.annotations.JsPackage;
import jsinterop.annotations.JsProperty;
import jsinterop.annotations.JsType;
import jsinterop.base.Js;
import jsinterop.base.JsPropertyMap;
import org.treblereel.gwt.three4g.textures.Texture;
import org.treblereel.gwt.three4g.math.Vector3;

@JsType(
    isNative = true,
    name = "THREE.MeshDistanceMaterialParameters",
    namespace = JsPackage.GLOBAL)
public interface MeshDistanceMaterialParameters extends MaterialParameters {
  @JsOverlay
  static MeshDistanceMaterialParameters create() {
    return Js.uncheckedCast(JsPropertyMap.of());
  }

  @JsProperty
  Texture getAlphaMap();

  @JsProperty
  double getDisplacementBias();

  @JsProperty
  Texture getDisplacementMap();

  @JsProperty
  double getDisplacementScale();

  @JsProperty
  double getFarDistance();

  @JsProperty
  Texture getMap();

  @JsProperty
  double getNearDistance();

  @JsProperty
  Vector3 getReferencePosition();

  @JsProperty
  void setAlphaMap(Texture alphaMap);

  @JsProperty
  void setDisplacementBias(double displacementBias);

  @JsProperty
  void setDisplacementMap(Texture displacementMap);

  @JsProperty
  void setDisplacementScale(double displacementScale);

  @JsProperty
  void setFarDistance(double farDistance);

  @JsProperty
  void setMap(Texture map);

  @JsProperty
  void setNearDistance(double nearDistance);

  @JsProperty
  void setReferencePosition(Vector3 referencePosition);
}
